unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,108 +23,181 @@ interface UpdateReferralFeeProps {
-        <ToggleSwitch
-          title="Referrer fee %"
-          enabled={referralFee}
-          disabled={disabledInput}
-          setEnabled={(enabled) => {
-            setReferralFee(enabled)
-            setValue(
-              'referralFeePercentage',
-              enabled ? (referralFeePercentage ?? 0) / 100 : 0
-            )
-          }}
-        />
         <Input
           type="number"
           {...register('referralFeePercentage', {
             valueAsNumber: true,
-            min: 0,
+            min: 1,
             max: 100,
           })}
           error={
             errors?.referralFeePercentage &&
-            'This field accept percentage value between 0 and 100.'
+            'This field accept percentage value between 1 and 100.'
           }
-          disabled={disabledInput || !referralFee}
+          disabled={isDisabledReferrerInput}
         />
       </div>
+
+      <div className="grid gap-2">
+        <ToggleSwitch
+          title="Referrer address"
+          description={
+            'If no address is provided, the default referral fee will apply to any referrer.'
+          }
+          enabled={isReferralAddressToggled}
+          disabled={isDisabledReferrerInput}
+          setEnabled={(enabled: boolean) => {
+            setIsReferralAddressToggled(enabled)
+          }}
+        />
+
+        <AddressInput
+          label=""
+          value={referralAddress}
+          disabled={isDisabledReferrerInput || !isReferralAddressToggled}
+          onChange={(value: any) => {
+            setValue('referralAddress', value)
+          }}
+          error={errors?.referralAddress?.message}
+          onResolveName={onResolveName}
+        />
+      </div>
+
       {isManager && (
         <Button
           className="w-full md:w-1/3"
           type="submit"
-          disabled={disabledInput}
-          loading={setReferrerFeeMutation.isLoading}
+          disabled={
+            isDisabledReferrerInput ||
+            !referralFeePercentage ||
+            !!(referralAddress && !isValidAddress)
+          }
+          loading={isSettingReferrerFee}
         >
           Apply
         </Button>
       )}
+
+      <div className="flex flex-col mt-5 items-start">
+        <div className="flex items-center justify-between w-full">
+          <Card.Title>Referrers ({referralFees.length})</Card.Title>
+
+          {!isLoading ? (
+            <Button
+              className="w-full max-w-[120px]"
+              type="button"
+              onClick={() => {
+                refetch()
+              }}
+              iconLeft={
+                <Image
+                  className="justify-self-left"
+                  width="20"
+                  height="20"
+                  alt="Apple Wallet"
+                  src={`/images/illustrations/refresh.svg`}
+                />
+              }
+            >
+              Refresh
+            </Button>
+          ) : null}
```

Comment:
In case the subgraph is not full in sync we can add a "Refresh" button which will refresh the referrer fees list. This can improve overall the UX experience 😁